File: ATDP01.py
import sys
n = int(input())
h = [0]+list(map(int, input().split()))

sys.setrecursionlimit(2*10**5+1)
res = [-1] * (n+1)
def f(n):
    if n == 1: return 0
if n == 2: return abs(h[2] - h[1])
    if res[n] != -1: return res[n]
    res[n] = min(f(n-1) + abs(h[n] - h[n-1]), f(n-2) + abs(h[n] - h[n-2]))
    return res[n]
print(f(n))

File: ATDP01.cpp
#include <bits/stdc++.h>
using namespace std;
int n, a[100005], f[100005];
int dp(int x){
    if (x == 1) return 0;
    if (x == 2) return abs(a[2] - a[1]);
    if (f[x] > 0) return f[x];
    f[x] = min(dp(x-1) + abs(a[x] - a[x-1]), dp(x-2) + abs(a[x] - a[x-2]));
    return f[x];
}
int main(){
    cin >> n;
    for (int i = 1; i <= n; ++i) cin >> a[i];
    cout << dp(n);
    return 0;
}